org.eclipse.dawnsci.remotedataset.client/src/org/eclipse/dawnsci/remotedataset/client/RemoteDataHolder.java

```diff
@@ -0,0 +1,193 @@
+		}
+	}
+
+	private Map<String, Object> getTreeMap() throws Exception {
+		
+		urlBuilder.setGet(true);
+		final URL url = new URL(urlBuilder.getTreeURL());
+		URLConnection  conn = url.openConnection();
+
+		final BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"));
+		try {
+			String xml = reader.readLine();
+			IMarshallerService marshaller = new XMLMarshallerService();
+			return (Map<String, Object>)marshaller.unmarshal(xml, Map.class);
+
+		} finally {
+			reader.close();
+		}
+	}
+	
+	@Override
+	public IDataHolder clone() {
+		return null;
+	}
+
+	@Override
+	public IMetadata getMetadata() {
+		return null;
+	}
+
+	@Override
+	public void setMetadata(IMetadata md) {
+
+	}
+
+	@Override
+	public IDataset getDataset(int index) {
+		return null;
+	}
+
+	@Override
+	public IDataset getDataset(String name) {
+		return null;
+	}
+
+	@Override
+	public ILazyDataset getLazyDataset(int index) {
+		return null;
+	}
+
+	@Override
+	public ILazyDataset getLazyDataset(String name) {
+		URLBuilder b = new URLBuilder(urlBuilder);
+		b.setDataset(name);
+		RemoteLoader loader = new RemoteLoader(b);
+		try {
+			RemoteDatasetInfo datasetInfo = loader.getDatasetInfo();
+			LazyDynamicDataset lazy = new LazyDynamicDataset(name, datasetInfo.getDtype(),
+					datasetInfo.getIsize(), datasetInfo.getShape(),
+					datasetInfo.getMaxShape(), loader);
+			
+			return lazy;
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
+		return null;
+	}
+
+	@Override
+	public boolean contains(String name) {
+		return false;
+	}
+
+	@Override
+	public String[] getNames() {
+		return null;
+	}
+
+	@Override
+	public String getName(int index) {
+		return null;
+	}
+
+	@Override
+	public int size() {
+		return 0;
+	}
+
+	@Override
+	public int namesSize() {
+		return 0;
+	}
+
+	@Override
+	public String getFilePath() {
+		return path;
+	}
+
+	@Override
+	public void setFilePath(String path) {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public boolean addDataset(String name, ILazyDataset dataset) {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public Map<String, ILazyDataset> toLazyMap() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public void setLoaderClass(Class<? extends IFileLoader> clazz) {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public Class<? extends IFileLoader> getLoaderClass() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public void clear() {
+		// TODO Auto-generated method stub
+
+	}
+
+	@Override
+	public List<ILazyDataset> getList() {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public Tree getTree() {
+		// TODO Auto-generated method stub
+		return tree;
+	}
+
+	@Override
+	public void setTree(Tree tree) {
+		throw new RuntimeException("Cannot overwrite tree in remote data holder!");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)